### ISSUE TITLE
OnActionEnded is raised twice for the Select Input Action in Editor

### ISSUE THREAD

**Issue by bowlingforsoap** [microsoft/MixedRealityToolkit-Unity#8776]
## Describe the bug

OnActionEnded is raised twice for the Select Input Action in Editor. This happens, if you perform a click with the Input Simulation being in the Gesture Hand Simulation mode.

## To reproduce

1. Create a MonoBehaviour script and implement the IMixedRealityInputActionHandler in it:

        private void Start()
        {
            CoreServices.InputSystem?.RegisterHandler<IMixedRealityInputActionHandler>(this);
        }

        private void OnDestroy()
        {
            CoreServices.InputSystem?.UnregisterHandler<IMixedRealityInputActionHandler>(this);

        }

        public void OnActionStarted(BaseInputEventData eventData) {}

        public void OnActionEnded(BaseInputEventData eventData)
        {
            if (eventData.MixedRealityInputAction == selectAction)
            {
                Debug.Log($"OnActionEnded raised: {Time.time}");
            }
        }

2. Set _selectAction_ to Select in Inspector
3. Make sure that MixedRealityToolkit -> Input -> Input Data Providers -> Input Simulation Service -> Default Hand Simulation Mode is set to _Gestures_
4. Press Play in Unity
5. When you click with the LMB you will receive the OnActionEnded twice.

## Expected behavior

OnActionEnded is raised only once for one click.

## Screenshots

Console output after 2 LMB clicks:
![image](https://user-images.githubusercontent.com/7228421/95585637-c2a8ec80-0a3f-11eb-8ecb-21fe33081ace.png)

## Your setup (please complete the following information)

- Unity Version [e.g. 2019.4.10f1]
- MRTK Version [e.g. v2.4]

## Target platform (please complete the following information)

- Editor, with UWP selected in the build settings

## Additional context

I found that the two callbacks have different call stacks:
* the first one comes from the MixedRealityInputSystem.RaiseInputUp
![image](https://user-images.githubusercontent.com/7228421/95585840-19aec180-0a40-11eb-89f3-a964247f7048.png)
* the second one comes from MixedRealityInputSystem.RaiseGestureCompleted
![image](https://user-images.githubusercontent.com/7228421/95585879-29c6a100-0a40-11eb-9a7f-ce4b69c5d61f.png)

It seems that both, lines 151 and 154 in _SimulatedGestureHand.cs_ are causing the input action to be raised:
![image](https://user-images.githubusercontent.com/7228421/95586054-66929800-0a40-11eb-8c31-882c7563dffb.png)

Additionally, I don't think that this is only the case for the Select Input Action. I first encountered the problem, when I wasn't even checking, which MixedRealityInputAction caused OnActionEnded.

**Comment by keveleigh** [comment]
Closing in favor of https://github.com/microsoft/MixedRealityToolkit-Unity/issues/7733

>The OnInputAction* events gather all other fired input events and send them along. Currently, with the default legacy profiles, we support the "select"/"tap" gesture, which raises a gesture completed event at the time of recognition. This would be why you aren't seeing the double event when holding longer, since "select" has a time component and doesn't fire if that timespan has elapsed.
>
>Basically, you're seeing OnGestureCompleted and OnInputUp, both with "Select" actions, coming across the OnInputActionEnded event. I'd recommend removing select from the gesture profile if this is an issue.

**Comment by keveleigh** [comment]
Though, we may have an additional issue with making the simulated hand respect that gesture profile

**Comment by IssueSyncBot** [comment]
We appreciate your feedback and thank you for reporting this issue.

Microsoft Mixed Reality Toolkit version 2 (MRTK2) is currently in limited support. This means that Microsoft is only fixing high priority security issues. Unfortunately, this issue does not meet the necessary priority and will be closed. If you strongly feel that this issue deserves more attention, please open a new issue and explain why it is important.

Microsoft recommends that all new HoloLens 2 Unity applications use MRTK3 instead of MRTK2.

Please note that MRTK3 was released in August 2023. It features an all-new architecture for developing rich mixed reality experiences and has a minimum requirement of Unity 2021.3 LTS. For more information about MRTK3, please visit [https://www.mixedrealitytoolkit.org](https://www.mixedrealitytoolkit.org).

Thank you for your continued support of the Mixed Reality Toolkit!
